### ISSUE TITLE
The problem implementing print in gpython.

### ISSUE THREAD

**Issue by Sungmin-Joo** [go-python/gpython#91]
Even if sep is not required, the option must be inserted.

**gpython**  
```
with open("testfile","w") as f:
    print("1234", end = "@@@", file = f)
with open("testfile","r") as f:
    print(f.read())`

Traceback (most recent call last):
  File "test.py", line 3, in <module>
    FIXME line of source goes here
TypeError: 'print() argument 2 must be str, not file'
2019/09/26 11:21:21 TypeError: 'print() argument 2 must be str, not file'
```  

**when fixed code**
```
with open("testfile","w") as f:
    print("1234", sep = '', end = "@@@", file = f)
with open("testfile","r") as f:
    print(f.read())

1234@@@
```

**Comment by ncw** [comment]
The gpython `print` is fairly minimal at the moment and certainly needs improving.

I think this particular bug is caused by this line doing the argument parsing wrong (not sure exactly why).

https://github.com/go-python/gpython/blob/4b347aa2bb5acd7fdd2fcb9a52b463f6c5535740/builtin/builtin.go#L185

**Comment by Sungmin-Joo** [comment]
@ncw 
There is a temporary way to modify the build_print.
But I don't think it's the right way.
```
func builtin_print(self py.Object, args py.Tuple, kwargs py.StringDict) (py.Object, error) {
    var (
        sepObj py.Object = py.String(" ")
        endObj py.Object = py.String("\n")
        file   py.Object = py.MustGetModule("sys").Globals["stdout"]
        flush  py.Object
    )
    kwlist := []string{"sep", "end", "file", "flush"}
    for _, kw := range kwlist {
        switch _, ok := kwargs[kw]; ok {
        case true:
            continue
        default:
            if kw == "sep" {
                kwargs[kw] = sepObj
            } else if kw == "end" {
                kwargs[kw] = endObj
            } else if kw == "file" {
                kwargs[kw] = file
            } else {
                kwargs[kw] = flush
            }
        }
    }
    err := py.ParseTupleAndKeywords(nil, kwargs, "|ssOO:print", kwlist, &sepObj, &endObj, &file, &flush)
    ...
```

**Comment by ncw** [comment]
`py.ParseTupleAndKeywords` should be doing that for us...

The bug must be in this bit of code I think...  No time to investigate further right now - over to you to have a look!

https://github.com/go-python/gpython/blob/8c361a8eb99395b87df36de243df7f4d7b8d07d9/py/args.go#L444-L453

**Comment by Sungmin-Joo** [comment]
when will kewordonly be used?
Could you delete kewordonly and let me put the nil in when ok is "false"?

In this way.
```
for i, kw := range kwlist {
    if value, ok := kwargs[kw]; ok {
        if len(args) > i {
            return ExceptionNewf(TypeError, "%s() got multiple values for argument '%s'", name, kw)
        }
        args = append(args, value)
    } else  {
        args = append(args,nil)
    }
}
```

**Comment by drew-512** [comment]
I've fixed this and expanded the print coverage tests and is looking good.  See [7d2a8d9](https://github.com/go-python/gpython/pull/144/commits/7d2a8d9a28639565862ca5ff915f3697d294e6cb).  

Hopefully, we'll see this merged shortly via [PR144](https://github.com/go-python/gpython/pull/144)

**Comment by drew-512** [comment]
suggest @sbinet close -- fixed w/ https://github.com/go-python/gpython/pull/151

**Comment by sbinet** [comment]
fixed w/ #151.
